src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -254,6 +366,20 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
             {
                 AsyncCompletionLogger.LogExpanderUsage();
 
+                // It's possible we didn't provide expanded items at the beginning of completion session because it was slow even if the feature is enabled.
+                // ExpandedItemsTask would be available in this case, so we just need to return its result.
+                if (session.Properties.TryGetProperty(ExpandedItemsTask, out Task<(VSCompletionContext, CompletionList)> expandedItemsTask))
+                {
+                    // Make sure the task is removed when returning expanded items,
+                    // so duplicated items won't be added in subsequent list updates.
+                    session.Properties.RemoveProperty(ExpandedItemsTask);
+                    var (expandedContext, expandedCompletionList) = await expandedItemsTask.ConfigureAwait(false);
+                    AddPropertiesToSession(session, expandedCompletionList, triggerLocation: null);
```

Comment:
why is the trigger location null?

Reply:
Since it's already saved in the property bag so we don't need to do it again.
That said, there's already guard in the help method and this seems to just make thing even more confusing, so I changed it to always pass in a trigger location instead.